apiserver/facades/client/cloud/cloud_test.go

```diff
@@ -1291,16 +1749,33 @@ func (s *cloudSuite) TestModifyCloudAccess(c *gc.C) {
 		},
 	})
 	c.Assert(err, jc.ErrorIsNil)
-	s.backend.CheckCallNames(c, "Cloud", "ControllerTag", "CreateCloudAccess", "Cloud", "ControllerTag", "RemoveCloudAccess")
-	s.backend.CheckCall(c, 2, "CreateCloudAccess", "fluffy", names.NewUserTag("fred"), permission.AddModelAccess)
-	s.backend.CheckCall(c, 5, "RemoveCloudAccess", "fluffy", names.NewUserTag("mary"))
 	c.Assert(results.Results, gc.DeepEquals, []params.ErrorResult{
 		{}, {},
 	})
 }
 
 func (s *cloudSuite) TestModifyCloudUpdateAccess(c *gc.C) {
-	s.backend.cloudAccess = permission.AddModelAccess
+	adminTag := names.NewUserTag("admin")
+	defer s.setup(c, adminTag).Finish()
+
+	cloud := jujucloud.Cloud{
+		Name:      "fluffy",
+		Type:      "fluffy",
+		AuthTypes: []jujucloud.AuthType{jujucloud.EmptyAuthType, jujucloud.UserPassAuthType},
+		Regions:   []jujucloud.Region{{Name: "nether", Endpoint: "endpoint"}},
+	}
+
+	fredTag := names.NewUserTag("fred")
+
+	backend := s.backend.EXPECT()
+	backend.Cloud("fluffy").Return(cloud, nil)
+	backend.CreateCloudAccess("fluffy", fredTag,
+		permission.AdminAccess).Return(nil).MaxTimes(1)
+	backend.GetCloudAccess("fluffy", fredTag).Return(permission.AddModelAccess,
+		nil).MaxTimes(1)
+	backend.UpdateCloudAccess("fluffy", fredTag,
+		permission.AdminAccess).Return(nil).MaxTimes(1)
```

Comment:
MaxTimes inside specific tests is also a specific test is also a red flag. There's a good chance these aren't called at all, in which case their presence here wrong.

Especially since these functions look like these methods mutate state

Reply:
Without these `EXPECTs` the tests fails.
```
FAIL: cloud_test.go:1757: cloudSuite.TestModifyCloudUpdateAccess

cloud_test.go:1779:
    results, err := s.api.ModifyCloudAccess(params.ModifyCloudAccessRequest{
        Changes: []params.ModifyCloudAccess{
            {
                Action:   params.GrantCloudAccess,
                CloudTag: names.NewCloudTag("fluffy").String(),
                UserTag:  names.NewUserTag("fred").String(),
                Access:   "admin",
            },
        },
    })
/home/capricorn/code/go/pkg/mod/github.com/golang/mock@v1.6.0/gomock/controller.go:183:
    h.t.Fatalf(format, args...)
... Error: Unexpected call to *mocks.MockBackend.CreateCloudAccess([fluffy user-fred admin]) at /home/capricorn/code/go/src/juju/apiserver/facades/client/cloud/cloud.go:1283 because: there are no expected calls of the method "CreateCloudAccess" for that receiver

```

Reply:
it looks like you have dropped the line entirely, since the error is an `Unexpected call`. I meant only the call to `MaxTimes(1)`, in case that wasn't clear

You should decide whether or not these lines should be present or not, based on an understanding behind the intention of the test. And either include it without `MaxTimes(1)`, or drop the line entirely